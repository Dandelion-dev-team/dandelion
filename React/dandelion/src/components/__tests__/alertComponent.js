import React from "react"
import renderer from "react-test-renderer"
import AlertComponent from "../alertComponent"

describe("Alert", () => {
  it("renders correctly", () => {
    const testProp = { 'id': '1', 'description': 'Test', 'updated_date': '2022-02-19T17:05:22.178Z', "status": "amber" };
    const tree = renderer
      .create(<AlertComponent alert={testProp}/>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})