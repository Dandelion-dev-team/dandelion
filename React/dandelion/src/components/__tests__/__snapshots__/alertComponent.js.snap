// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert renders correctly 1`] = `
<div
  className="alertContainer"
>
  <div
    className="top-divider"
  >
    <hr />
  </div>
  <div
    className="row"
  >
    <div
      className="text-content"
    >
      <h3>
        Test
      </h3>
      <div
        className="date-recorded"
      >
        <h3>
          Sat Feb 19 2022
        </h3>
      </div>
    </div>
    <img
      src="test-file-stub"
    />
    <div
      className="clear-alert"
    >
      <style
        dangerouslySetInnerHTML={
          Object {
            "__html": ".css-i4bv87-MuiSvgIcon-root{-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;width:1em;height:1em;display:inline-block;fill:currentColor;-webkit-flex-shrink:0;-ms-flex-negative:0;flex-shrink:0;-webkit-transition:fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;font-size:1.5rem;}",
          }
        }
        data-emotion="css i4bv87-MuiSvgIcon-root"
      />
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="ClearIcon"
        focusable="false"
        onClick={[Function]}
        viewBox="0 0 24 24"
      >
        <path
          d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
        />
      </svg>
    </div>
  </div>
  <div
    className="bottom-divider"
  >
    <hr />
  </div>
</div>
`;

exports[`Header renders correctly 1`] = `
<div
  className="alertContainer"
>
  <div
    className="top-divider"
  >
    <hr />
  </div>
  <div
    className="row"
  >
    <div
      className="text-content"
    >
      <h3>
        Test
      </h3>
      <div
        className="date-recorded"
      >
        <h3>
          Sat Feb 19 2022
        </h3>
      </div>
    </div>
    <img
      src="test-file-stub"
    />
    <div
      className="clear-alert"
    >
      <style
        dangerouslySetInnerHTML={
          Object {
            "__html": ".css-i4bv87-MuiSvgIcon-root{-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;width:1em;height:1em;display:inline-block;fill:currentColor;-webkit-flex-shrink:0;-ms-flex-negative:0;flex-shrink:0;-webkit-transition:fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;font-size:1.5rem;}",
          }
        }
        data-emotion="css i4bv87-MuiSvgIcon-root"
      />
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="ClearIcon"
        focusable="false"
        onClick={[Function]}
        viewBox="0 0 24 24"
      >
        <path
          d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
        />
      </svg>
    </div>
  </div>
  <div
    className="bottom-divider"
  >
    <hr />
  </div>
</div>
`;
